File: 235. Lowest Common Ancestor of a Binary Search Tree/Java/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        while (root != null) {
            if ((p.val < root.val && q.val > root.val) || (p.val > root.val && q.val < root.val) || p.val == root.val || q.val == root.val) {
                return root;
            }
            if (p.val < root.val && q.val < root.val) {
                root = root.left;
            }
            if (p.val > root.val && q.val > root.val) {
                root = root.right;
            }
        }
        return null;
    }
}

File: 235. Lowest Common Ancestor of a Binary Search Tree/Python/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):
    def lowestCommonAncestor(self, root, p, q):
        """
        :type root: TreeNode
        :type p: TreeNode
        :type q: TreeNode
        :rtype: TreeNode
        """
        while root:
            if (p.val < root.val and q.val > root.val) or (p.val > root.val and q.val < root.val) or (
                    p.val == root.val) or (q.val == root.val):
                return root
            if p.val < root.val and q.val < root.val:
                root = root.left
            if p.val > root.val and q.val > root.val:
                root = root.right
